File: examples/python/NeuralNetwork/neural_network.py
#!/usr/bin/env python3
import cv2
import depthai as dai
import numpy as np
import time

# Create pipeline
with dai.Pipeline() as pipeline:
    cameraNode = pipeline.create(dai.node.Camera).build()
    # Longer form - useful in case of a local NNArchive
    # modelDescription = dai.NNModelDescription("yolov6-nano", platform=pipeline.getDefaultDevice().getPlatformAsString())
    # archive = dai.NNArchive(dai.getModelFromZoo(modelDescription))
    # neuralNetwork = pipeline.create(dai.node.NeuralNetwork).build(cameraNode, archive)
    neuralNetwork = pipeline.create(dai.node.NeuralNetwork).build(cameraNode, dai.NNModelDescription("yolov6-nano"))

    qNNData = neuralNetwork.out.createOutputQueue()

    pipeline.start()


    while pipeline.isRunning():
        inNNData: dai.NNData = qNNData.get()
        tensor = inNNData.getFirstTensor()
        assert(isinstance(tensor, np.ndarray))
        print(f"Received NN data: {tensor.shape}")

File: examples/cpp/NeuralNetwork/neural_network.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

#include "depthai/depthai.hpp"

int main() {
    // Create pipeline
    dai::Pipeline pipeline;

    // Create nodes
    auto cameraNode = pipeline.create<dai::node::Camera>();
    auto neuralNetwork = pipeline.create<dai::node::NeuralNetwork>();

    // Build nodes
    cameraNode->build();
    // Longer form - useful in case of a local NNArchive
    // auto modelDescription = dai::NNModelDescription("yolov6-nano", pipeline.getDefaultDevice()->getPlatformAsString());
    // auto archive = dai::NNArchive(dai::getModelFromZoo(modelDescription));
    // neuralNetwork->build(cameraNode, archive);
    dai::NNModelDescription modelDescription;
    modelDescription.model = "yolov6-nano";
    modelDescription.platform = pipeline.getDefaultDevice()->getPlatformAsString();
    neuralNetwork->build(cameraNode, modelDescription);

    // Create output queue
    auto qNNData = neuralNetwork->out.createOutputQueue();

    // Start pipeline
    pipeline.start();

    // Main loop
    while(pipeline.isRunning()) {
        auto inNNData = qNNData->get<dai::NNData>();
        auto tensor = inNNData->getFirstTensor<float>();
        std::cout << "Received NN data: " << tensor.shape()[0] << "x" << tensor.shape()[1] << std::endl;
    }
    return 0;
}